elixir/apps/domain/lib/domain/auth/adapters/microsoft_entra/api_client.ex

```diff
@@ -0,0 +1,158 @@
+defmodule Domain.Auth.Adapters.MicrosoftEntra.APIClient do
+  use Supervisor
+
+  @pool_name __MODULE__.Finch
+
+  def start_link(_init_arg) do
+    Supervisor.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  @impl true
+  def init(_init_arg) do
+    children = [
+      {Finch,
+       name: @pool_name,
+       pools: %{
+         default: pool_opts()
+       }}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp pool_opts do
+    transport_opts =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:finch_transport_opts)
+
+    [conn_opts: [transport_opts: transport_opts]]
+  end
+
+  def list_users(api_token) do
+    endpoint =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:endpoint)
+
+    uri =
+      URI.parse("#{endpoint}/v1.0/users")
+      |> URI.append_query(
+        URI.encode_query(%{
+          "$select" =>
+            Enum.join(
+              ~w[
+                id
+                accountEnabled
+                displayName
+                givenName
+                surname
+                mail
+                userPrincipalName
+              ],
+              ","
+            ),
+          "$filter" => "accountEnabled eq true"
```

Comment:
I think it would be good to have some flexibility in what gets pulled/synced from the different IDPs and we probably do want to make this some type of variable rather than a hard coded string.  I'll move this to a module attribute.

Reply:
I think in future we should just add some UI for the provider settings and allow them to extend the filters, but `accountEnabled` should be always true, there is no point to sync trash data for us - they will never be able to sign in.